app/assets/javascripts/pages/simulation.js

```diff
@@ -1,39 +1,94 @@
-import "/codemirror-mode.js";
-import AlertDisplay from "/alert-display.js";
-import newModel from "/new-model.js";
-import Tortoise from "/beak/tortoise.js";
-
-var loadingOverlay  = document.getElementById("loading-overlay");
-var activeContainer = loadingOverlay;
-var modelContainer  = document.querySelector("#netlogo-model-container");
-var nlogoScript     = document.querySelector("#nlogo-code");
-
-var pageTitle       = function(modelTitle) {
-  if (modelTitle != null && modelTitle != "") {
-    return "NetLogo Web: " + modelTitle;
+import { createNotifier, listenerEvents } from "/notifications/listener-events.js"
+import { createDebugListener } from "/notifications/debug-listener.js"
+import { createIframeRelayListener } from "/notifications/iframe-relay-listener.js"
+import { attachQueryHandler } from "/queries/iframe-query-handler.js"
+
+import { fakeStorage, NamespaceStorage } from "/namespace-storage.js"
+import { WipListener } from "/beak/wip-listener.js"
+
+import "/codemirror-mode.js"
+import AlertDisplay from "/alert-display.js"
+import newModel from "/new-model.js"
+import Tortoise from "/beak/tortoise.js"
+
+var loadingOverlay  = document.getElementById('loading-overlay')
+var activeContainer = loadingOverlay
+var modelContainer  = document.querySelector('#netlogo-model-container')
+var nlogoScript     = document.querySelector('#nlogo-code')
+
+const params    = new URLSearchParams(window.location.search)
+const paramKeys = Array.from(params.keys())
+if (paramKeys.length === 1) {
+  const maybeUrl = paramKeys[0]
+  if (maybeUrl.startsWith('http') && params.get(maybeUrl) === '') {
+    params.delete(maybeUrl)
+    params.set('url', maybeUrl)
+    window.location.search = params.toString()
+  }
+}
+
+const listeners = []
+
+const disableWorkInProgress = params.has('disableWorkInProgress')
+var wipListener       = null
+var getWorkInProgress = null
```

Comment:
One option here is change this to:

```javascript
const [wipListener, getWorkInProgress] = (() => {
  ...
  return [wl, gwip];
})();
```

Another option is to *not*, and just leave things how they are.  But I'm just speaking for Camp No-`var`s (Especially in Top-Level Scopes).